### PR TITLE
Benchmark scale

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add a CMAKE variable to control the scaling of the benchmark tests.

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/echoMessageBenchmarks.cpp  1
- benchmarks/helics/filterBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
+ benchmarks/helics/echoBenchmarks.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

File: benchmarks/helics/echoBenchmarks.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ BENCHMARK_CAPTURE(BMecho_multiCore, tcpssCore, core_type::TCP_SS)
 // Register the UDP benchmarks
 BENCHMARK_CAPTURE(BMecho_multiCore, udpCore, core_type::UDP)
     ->RangeMultiplier(2)
-    ->Range(1, 1 << 4)
+    ->Range(1, maxscale)
```
[the remark this answers is not in the document]
Comment (phlptp) [reply benchmarks/helics/echoBenchmarks.cpp:174]: I will look into it but it doesn't look like it was before.

File: benchmarks/helics/timingBenchmarks.cpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ BENCHMARK_CAPTURE(BMtiming_multiCore, tcpssCore, core_type::TCP_SS)
 // Register the UDP benchmarks
 BENCHMARK_CAPTURE(BMtiming_multiCore, udpCore, core_type::UDP)
     ->RangeMultiplier(2)
-    ->Range(1, 1 << 4)
+    ->Range(1, maxscale)
```
[the remark this answers is not in the document]
Comment (phlptp) [reply benchmarks/helics/timingBenchmarks.cpp:176]: again wasn't before, but it maybe should have been

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/echoMessageBenchmarks.cpp  1
- benchmarks/helics/filterBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
+ benchmarks/helics/echoBenchmarks.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/filterBenchmarks.cpp  1
- benchmarks/helics/echoMessageBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
+ benchmarks/helics/echoBenchmarks.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/filterBenchmarks.cpp  1
- benchmarks/helics/echoMessageBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks.cpp  -1
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/echoMessageBenchmarks.cpp  1
- benchmarks/helics/filterBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks.cpp  -1
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

File: benchmarks/helics/echoBenchmarks.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (phlptp) [reply benchmarks/helics/echoBenchmarks.cpp:176]: What I tried to do was just assuming no other changes,  this instance was just using a fixed number so I made it use maxscale which was the same as the constant previously.    I think there were a few cases where in the UDP core it was using maxscale/2 but I don't think this was one of them based on the change line.

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/filterBenchmarks.cpp  1
- benchmarks/helics/echoMessageBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks.cpp  -1
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/echoMessageBenchmarks.cpp  1
- benchmarks/helics/filterBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks.cpp  -1
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

File: benchmarks/helics/echoBenchmarks.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (nightlark) [reply benchmarks/helics/echoBenchmarks.cpp:176]: You're right, these cases are either the same or lower than what they were before depending on the processor count -- I think it's just the phold change that was the equivalent of `maxscale/2` before the change.

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- benchmarks/helics/echoMessageBenchmarks.cpp  1
- benchmarks/helics/filterBenchmarks.cpp  1
         

Clones removed
==============
+ benchmarks/helics/echoBenchmarks_c.cpp  -2
+ benchmarks/helics/echoBenchmarks.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5456875&bid=17887396)

File: benchmarks/helics/pholdBenchmarks.cpp [hunk 3]
```diff
@@ -189,7 +189,7 @@ BENCHMARK_CAPTURE(BMphold_multiCore, tcpssCore, core_type::TCP_SS)
 // Register the UDP benchmarks
 BENCHMARK_CAPTURE(BMphold_multiCore, udpCore, core_type::UDP)
     ->RangeMultiplier(2)
-    ->Range(1, 1 << 4)
+    ->Range(1, maxscale)
```
[the remark this answers is not in the document]
Comment (phlptp) [reply benchmarks/helics/pholdBenchmarks.cpp]: I updated this one to match now

**Pull request merged**